pkg/component/shootsystem/shootsystem.go

```diff
@@ -305,6 +305,12 @@ func (s *shootSystem) getServiceAccountNamesToInvalidate() []string {
 			"resource-claim-controller")
 	}
 
+	if versionutils.ConstraintK8sGreaterEqual128.Check(s.values.Shoot.KubernetesVersion) {
```

Comment:
Should we not add a test for this similar to what we did in https://github.com/gardener/gardener/pull/7275/commits/ff962d055de9c84b767fa347d32f094e43f84b92?